theano/sandbox/gpuarray/__init__.py

```diff
@@ -64,8 +65,24 @@ def init_dev(dev, name=None):
     reg_context(name, context)
     pygpu_activated = True
     if config.print_active_device:
-        print("Mapped name %s to device %s: %s" % (name, dev, context.devname),
+        warn = None
+        cudnn_version = ""
+        if dev.startswith('cuda'):
+            try:
+                cudnn_version = dnn.version()
+                # 4100 should not print warning with cudnn 4 final.
+                if cudnn_version > 4100:
+                    warn = ("Your CuDNN version is more recent then Theano."
+                            " If you see problems, try updating Theano or"
+                            " downgrading CuDNN to version 4.")
+                cudnn_version = " (CuDNN version %s)" % cudnn_version
+            except Exception:
+                raise RuntimeError("CuDNN is mandatory with the GpuArray back-end.")
```

Comment:
We could also spend the one or two days required to pour CorrMM and then there would be a not-CPU fallback.

That would be the ideal situation.